vulture_os/authentication/user_portal/models.py

```diff
@@ -651,23 +651,21 @@ def to_html_template(self):
 
     def get_openid_callback_url(self, req_scheme, workflow_host, workflow_path, repo_id):
         if self.enable_external:
-            base_url = build_url("https" if self.external_listener.tls_profiles.count()>0 else "http", self.external_fqdn, self.external_listener.port)
+            base_url = build_url("https" if self.external_listener.has_tls() else "http", self.external_fqdn, '' if len(self.external_fqdn.split(':')) > 1 else self.external_listener.listener_set.first().port, '/')
```

Comment:
not the right correction: port shouldn't be included in the external FQDN!

Reply:
I understand what you mean but self.external_listener.port does not exist so there will be a problem if FQDN does not contain the port on IDP portals